### ISSUE TITLE
GCOV results for "osapi" obscured.

### ISSUE THREAD

**Issue by skliper** [nasa/osal#125]
The "make gcov" target for the "osapi" coverage test
feeds the output of GCOV through some SED scripting
that was originally constructed to handle GCOV output
for a single source file.

For the OSAPI case, however, GCOV is being handed data
that covers both the adapter source file and the original
source file.

The sequence of line joins and line edits provided does
not produce correct output for this case: it does manage
to generate the coverage line for the adapter source, but
the heuristic deals badly with the remainder of the file.
It would have worked if the second "File" line happened
to fall on an odd line number.

**Comment by skliper** [comment]
_Imported from trac issue 102.  Created by glimes on 2015-10-19T12:16:03, last modified: 2015-12-22T16:20:20_

**Comment by skliper** [comment]
_Trac comment by glimes on 2015-10-19 16:38:34:_

Review code available from the `trac-102-parse-gcov-outputs`
branch on Babelfish at [changeset:80e36f3f]. This will need
to be included in, or rebased onto, the Trac-22 changes when
they are merged into the `development` branch.

**Comment by skliper** [comment]
_Trac comment by glimes on 2015-12-03 17:53:06:_

Everything changed -- all the Makefiles moved and only ut-osal/Makefile
still runs the gcov command -- but it still has the same bug.

The old commit can't be applied due to the changes but the change still
needs to be considered. Returning to "in_work" so I can fix it up.

**Comment by skliper** [comment]
_Trac comment by glimes on 2015-12-03 19:29:26:_

Not yet really tested, but [changeset:5b0f1e9 5b0f1e9 trac-102-parse-gcov-outputs] has
the change I want to make. Basically move the processing out of the fragile
sed pipeline into a simple awk script, then modify the makefile to use the
new awk script.

**Comment by skliper** [comment]
_Trac comment by glimes on 2015-12-03 19:41:56:_

Confirmed: yes, we still need this fix.

Typical Bamboo log without this change (from CFS-CFSOSAL54-244/log):

03-Dec-2015 12:09:23 	Unit Test Code Coverage Summary
03-Dec-2015 12:09:23 	
03-Dec-2015 12:09:23 	        Level  Lines  Source File Name
03-Dec-2015 12:09:23 	      -------  -----  ------------------------
03-Dec-2015 12:09:23 	      100.00%     22  $top/osal/src/os/vxworks6/osnetwork.c
03-Dec-2015 12:09:23 	       95.05%    101  $top/osal/src/unit-test-coverage/vxworks6/ut-osal/src/osapi.c
03-Dec-2015 12:09:23 	       99.17%    361  $top/osal/src/os/vxworks6/osfileapi.c
03-Dec-2015 12:09:23 	       93.10%    145  $top/osal/src/os/vxworks6/ostimer.c
03-Dec-2015 12:09:23 	

Typical bamboo log after this change:

03-Dec-2015 16:32:46 	Unit Test Code Coverage Summary
03-Dec-2015 16:32:46 	
03-Dec-2015 16:32:46 	        Level  Lines  Source File Name
03-Dec-2015 16:32:46 	      -------  -----  ------------------------
03-Dec-2015 16:32:46 	      100.00%     22  $top/osal/src/os/vxworks6/osnetwork.c
03-Dec-2015 16:32:46 	      100.00%    277  $top/osal/src/os/vxworks6/osfilesys.c
03-Dec-2015 16:32:46 	       95.05%    101  $top/osal/src/unit-test-coverage/vxworks6/ut-osal/src/osapi.c
03-Dec-2015 16:32:46 	       98.05%    718  $top/osal/src/os/vxworks6/osapi.c
03-Dec-2015 16:32:46 	       99.17%    361  $top/osal/src/os/vxworks6/osfileapi.c
03-Dec-2015 16:32:46 	      100.00%    132  $top/osal/src/os/vxworks6/osloader.c
03-Dec-2015 16:32:46 	       93.10%    145  $top/osal/src/os/vxworks6/ostimer.c
03-Dec-2015 16:32:46 	       83.33%     24  $top/osal/src/unit-test-coverage/vxworks6/ut-osal/src/ostimer.c
03-Dec-2015 16:32:46 	

Basically, without this change, we lose all the entries for files whose "File"
information did not happen to fall into an odd-numbered line of the gcov output;
before, this was only an issue for osapi (and maybe ostimer) as in all the
other cases, each file was the first file in the output of the gcov command.

Now, all of the data are reported by a single gcov command in ut-osal, and since
gcov happens (today) to print an odd number of lines, the sed pipe is losing
half of the results. ugh.

Time for me to advocate merging [changeset:5b0f1e9] ...

**Comment by skliper** [comment]
_Trac comment by glimes on 2015-12-22 16:20:20:_

Merged in 2015-12-08
